video/out/wayland_common.h

```diff
@@ -144,7 +144,7 @@ struct vo_wayland_state {
     /* DND */
     struct wl_data_device_manager *dnd_devman;
     struct wl_data_offer *dnd_offer;
-    enum mp_dnd_action dnd_action;
+    int dnd_action; // actually enum mp_dnd_action
```

Comment:
Should we just add a `DND_NONE` for -1? This seems like something that's easy to miss. Not sure if that guarantees it will always be signed or not.

Reply:
> Not sure if that guarantees it will always be signed or not.

not 100% sure but I think it should

> This seems like something that's easy to miss.

no other backend has this bug at least, because only wayland has the dnd spread over a bunch of functions like this

Reply:
>no other backend has this bug at least, because only wayland has the dnd spread over a bunch of functions like this

Yeah that's true. Adding another type would also require modifying the related functions so let's keep it simple.